cloud/pkg/devicecontroller/controller/downstream.go

```diff
@@ -377,8 +377,14 @@ func addDeviceInstanceAndProtocol(device *v1alpha2.Device, deviceProfile *types.
 	}
 
 	deviceInstance.Twins = device.Status.Twins
-	deviceInstance.DataProperties = device.Spec.Data.DataProperties
-	deviceInstance.DataTopic = device.Spec.Data.DataTopic
+
+	// add dataProperties and dataTopic
+	if !(len(device.Spec.Data.DataProperties) == 0 && len(device.Spec.Data.DataTopic) == 0) {
```

Comment:
```suggestion
	if len(device.Spec.Data.DataProperties > 0 || len(device.Spec.Data.DataTopic) > 0 {
```
This way may be clearer

Reply:
Thanks.